woocommerce/woocommerce-ios
# [Woo POS] If there are no cart changes, checkout should not trigger an order sync API request

Closes: #13236


## Description


- Syncing order only if there are differences between cart and synced order

## Testing information


- Launch app
- Go to Menu > Point of Sale
- Add an item to the cart
- Tap Checkout
- Tap Add More to go back to the product selector
- Without changing anything, tap Checkout - nothing is being loaded, no order sync is being done, totals are showing instantly since there were not changes in the cart
- Tap Add More
- Add few more items, for example have items in cart like this: Product A, Product B, Product A
- Tap Checkout - observe that the totals are shimmering (order is being synced) and totals are updated
- Tap Add More
- Clear cart and add products in order like this: Product B, Product A, Product A
- Tap Checkout - observe that nothing is being loaded, no order sync is being done, totals are showing instantly since there were not changes in the cart (order of items might be different but basically the same items are in the cart)

## Demo Video


https://github.com/user-attachments/assets/2910f1f4-f83a-46aa-956d-9a870a0a32bc

---
